### DISCUSSION TITLE
Does `@babel/preset-env` `useBuiltIns: "usage"` also considered those nested dependencies?

### DISCUSSION THREAD

**Discussion by nyngwang** [babel/babel#16223] [Q&A] [answered]
When I use babel to bundle my project where I have a dependency @babel/runtime, I also found that this dependency also depends on another dependency called regenerator-runtime by its package.json. But regenerator-runtime is not included in my own package.json. Can I call this nested dependency of my project? Another question is that when I use `useBuiltIns: "usage"`, would it consider the nested dependency as my usage as well?

**Comment by nicolo-ribaudo** [comment, marked as answer]
> When I use babel to bundle my project where I have a dependency @babel/runtime, I also found that this dependency also depends on another dependency called regenerator-runtime by its package.json. But regenerator-runtime is not included in my own package.json. Can I call this nested dependency of my project?

Yes, you don't need to worry about it :)

> Another question is that when I use useBuiltIns: "usage", would it consider the nested dependency as my usage as well?

No, it only considers code that your bundler is passing to Babel. You can run Babel on your dependencies, so that it can detect polyfills there. If you need help with this, feel free to share your bundler configuration :)
